### PR TITLE
Add ci action

### PR DESCRIPTION
**Pull request by Arun-Niranjan**
Now whenever someone creates a PR,  pymdp unit tests will run automatically using python 3.8 and the modules listed in requirements.txt in a clean test environment

Note this currently ignores the experimental folder.

I am treating any flake8 linting problems as warnings only for now - but ideally we find a linter we're happy with and eventually fail builds which don't pass the linter.

### CODE REVIEW

**Comment by Arun-Niranjan** [comment]
tagging @conorheins and @alec-tschantz for review :-)

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8]
```
Comment (Arun-Niranjan) [inline .github/workflows/python-package.yml:18]: interestingly, our unit tests fail on python 3.9 (I think you can see the failed [build logs](https://github.com/Arun-Niranjan/pymdp/pull/3/checks?sha=9b38878948bb6ed7be926a5a4ec18747c06ed775) for the previous commit on this branch)

> E     File "/home/runner/work/pymdp/pymdp/test/test_control.py", line 28
> p = np.random.rand((*([no] + n_states)))
> E                             ^
> E   SyntaxError: can't use starred expression here

I think this should be a straightforward fix, but it should be done in a separate PR.

Comment (conorheins) [reply .github/workflows/python-package.yml:18]: Thanks, nice find. I need to look at Python 3.9 to see what the exact restrictions are, but  the 'official' numpy.random documentation recommends using `np.random.random_sample()` when passing in variable shape tuples (see https://numpy.org/devdocs/reference/random/generated/numpy.random.rand.html), so I will change those across the board when we're initializing random uniform matrices.

File: .github/workflows/python-package.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Python package
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (Arun-Niranjan) [inline .github/workflows/python-package.yml:10]: this might need some tweaking, but the easiest way to experiment is to land this PR as it is and push/raise Rs

Comment (conorheins) [reply .github/workflows/python-package.yml:10]: Cool. So I merged your commit, pulled and then pushed a new change (just replacing those calls to `np.random.rand` that you found in `test_control`), but I didn't see anything reflecting that the tests were run.

Comment (Arun-Niranjan) [reply .github/workflows/python-package.yml:10]: hmm - if you are directly pushing to master I would expect those to run.

Although best practise is to not push directly to master, try fixing something else and creating a PR? We'll see if that runs.

Comment (conorheins) [reply .github/workflows/python-package.yml:10]: Yeah I've just been pushing because I've (embarrassingly) never really worked with separate feature branches / issuing pull requests - I've always just been pushing directly to master. I'll do that now though to test this out - it's high time I learned how to use PRs properly anyway :)

Comment (Arun-Niranjan) [reply .github/workflows/python-package.yml:10]: It is honestly wonderful once you get the hang of it :-) I highly recommend reading this: https://github.com/susam/gitpr#create-pull-request - it's what I've been using for contributing to pymdp so far!

Comment (conorheins) [reply .github/workflows/python-package.yml:10]: This looks like exactly what I need, thank you! So I just created a branch, made a tiny change to the control.py file and then made a pull request. It did all the tests, as intended :) Super cool
